rackcity/models/asset.py

```diff
@@ -60,10 +62,11 @@ def validate_hostname_uniqueness(value, asset_id, change_plan, related_asset):
 
 def validate_asset_number_uniqueness(value, asset_id, change_plan, related_asset):
     assets, assets_cp = get_assets_for_cp(change_plan.id)
-    matching_assets = assets_cp.filter(asset_number=value)
+    matching_assets = assets_cp.filter(asset_number=value,change_plan=change_plan)
```

Comment:
again assets_cp queryset is already filtered on change_plan so you shouldn't need that here - were you getting unexpected results?

Reply:
still wondering about this